docs/meshstack.customer-group-sync.md

```diff
@@ -4,3 +4,206 @@ title: meshCustomer Group Synchronisation
 ---
 
 meshStack supports importing users and groups into meshStack from a source that supports the LDAP protocol, for example, Microsoft Active Directory. The [`Simple Paged Results Control`](https://www.ietf.org/rfc/rfc2696.txt) is used to do a paginated LDAP query to fetch the users and groups. Because of this, it is required that the LDAP server supports the `Simple Paged Results Control`. Other than importing users and groups, meshStack also has the capability to assign a [customer role](./meshcloud.customer.md#assign-meshcustomer-roles) to the imported group. The entities read via LDAP are imported to meshStack via the [meshObject import API](./meshstack.api.md#meshobject-api).
+
+## Configuration Reference
+
+This section describes how to configure the LDAP group synchronization in meshStack.
+For easier reference the following sections break down the configuration model in multiple parts. The union of these
+defines the full configuration model.
+
+The basic parameters required to configure the connection to LDAP are as follows.
+
+<!--snippet:mesh.identityconnector.ldap.core#type-->
+
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Dhall Type-->
+```dhall
+let ConnectionConfiguration =
+    {-
+      url: The LDAP connection URL which would be of the form ldap://example.com:389
+      base:
+          The base from which searches should be performed, which would be of the form "dc=meshcloud,dc=io"
+      username: The username of the user who can connect to the LDAP source to perform searches
+      password: The password of the user above
+    -}
+      { url : Text, base : Text, username : Text, password : Secret }
+```
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+The configuration related to the collection of data via LDAP is as follows.
+
+<!--snippet:mesh.identityconnector.ldap.collector#type-->
+
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Dhall Type-->
+```dhall
+{-
+  pageSize: The size of a single page that will be returned as a result of an LDAP search
+  group/person:
+    a configuration of type LDAPFilter which will contain the information needed to search the LDAP source
+    for users and groups.
+-}
+  let CollectorConfiguration =
+        { pageSize : Natural, group : LdapFilter, person : LdapFilter }
+
+  in  CollectorConfiguration
+```
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+The LdapFilter mentioned above has the following form.
+
+<!--snippet:mesh.identityconnector.ldap.filter#type-->
+
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Dhall Type-->
+```dhall
+{-
+    attributes:
+        A comma separated list of attributes that should be returned per LDAP entry.
+        For example, "cn, uniqueMember, description"
```

Comment:
d: Ah okay was this changed? I think previously we just returned the whole entity. Was there a problem with it? If I would have not code insight I would have problems to understand this text, probably mention that the returned attributes are later used for transformation into meshObject properties so the purpose of it is clearer and I can think about what properties I need?

Reply:
We specified attributes since the beginning. There was no explicit decision for that. 

I agree that the connection to transform part should be made clear. An alternative would be to re-write the config to only specify the attributes in the transformation configuration and infer the needed attributes for spring via dhall. This way we eliminate configuration errors where attributes are expected in transformations and not specified in collector configuration.

Reply:
It was not changed by me. When I picked up the task it was already like that.

Reply:
@felixzieger we could do that but shouldn't be part of this PR I think. Please report a ticket if you think it would be needed so that it can be handled seperately.